File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/tag/create/TagCreation.js
/**
 * @author pkupczyk
 */
define([ "stjs" ], function(stjs) {
	var TagCreation = function() {
	};
	stjs.extend(TagCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.tag.create.TagCreation';
		constructor.serialVersionUID = 1;
		prototype.code = null;
		prototype.description = null;
		prototype.experimentIds = null;
		prototype.sampleIds = null;
		prototype.dataSetIds = null;
		prototype.materialIds = null;
		prototype.creationId = null;

		prototype.getCode = function() {
			return this.code;
		};
		prototype.setCode = function(code) {
			this.code = code;
		};
		prototype.getDescription = function() {
			return this.description;
		};
		prototype.setDescription = function(description) {
			this.description = description;
		};
		prototype.getExperimentIds = function() {
			return this.experimentIds;
		};
		prototype.setExperimentIds = function(experimentIds) {
			this.experimentIds = experimentIds;
		};
		prototype.getSampleIds = function() {
			return this.sampleIds;
		};
		prototype.setSampleIds = function(sampleIds) {
			this.sampleIds = sampleIds;
		};
		prototype.getDataSetIds = function() {
			return this.dataSetIds;
		};
		prototype.setDataSetIds = function(dataSetIds) {
			this.dataSetIds = dataSetIds;
		};
		prototype.getMaterialIds = function() {
			return this.materialIds;
		};
		prototype.setMaterialIds = function(materialIds) {
			this.materialIds = materialIds;
		};
		prototype.getCreationId = function() {
			return this.creationId;
		};
		prototype.setCreationId = function(creationId) {
			this.creationId = creationId;
		};
	}, {
		experimentIds : {
			name : "List",
			arguments : [ "Object" ]
		},
		sampleIds : {
			name : "List",
			arguments : [ "Object" ]
		},
		dataSetIds : {
			name : "List",
			arguments : [ "Object" ]
		},
		materialIds : {
			name : "List",
			arguments : [ "Object" ]
		},
		creationId : "CreationId"
	});
	return TagCreation;
})

File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/tag/create/TagCreation.java
/*
 * Copyright 2013 ETH Zuerich, CISD
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.tag.create;

import java.util.List;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.CreationId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.interfaces.ICreationIdHolder;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.dataset.id.IDataSetId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.experiment.id.IExperimentId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.material.id.IMaterialId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.sample.id.ISampleId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.tag.create.TagCreation")
public class TagCreation implements ICreation, ICreationIdHolder, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    private String code;

    private String description;

    private List<? extends IExperimentId> experimentIds;

    private List<? extends ISampleId> sampleIds;

    private List<? extends IDataSetId> dataSetIds;

    private List<? extends IMaterialId> materialIds;

    private CreationId creationId;

    public String getCode()
    {
        return code;
    }

    public void setCode(String code)
    {
        this.code = code;
    }

    public String getDescription()
    {
        return description;
    }

    public void setDescription(String description)
    {
        this.description = description;
    }

    public List<? extends IExperimentId> getExperimentIds()
    {
        return experimentIds;
    }

    public void setExperimentIds(List<? extends IExperimentId> experimentIds)
    {
        this.experimentIds = experimentIds;
    }

    public List<? extends ISampleId> getSampleIds()
    {
        return sampleIds;
    }

    public void setSampleIds(List<? extends ISampleId> sampleIds)
    {
        this.sampleIds = sampleIds;
    }

    public List<? extends IDataSetId> getDataSetIds()
    {
        return dataSetIds;
    }

    public void setDataSetIds(List<? extends IDataSetId> dataSetIds)
    {
        this.dataSetIds = dataSetIds;
    }

    public List<? extends IMaterialId> getMaterialIds()
    {
        return materialIds;
    }

    public void setMaterialIds(List<? extends IMaterialId> materialIds)
    {
        this.materialIds = materialIds;
    }

    @Override
    public CreationId getCreationId()
    {
        return creationId;
    }

    public void setCreationId(CreationId creationId)
    {
        this.creationId = creationId;
    }

}
